docs/modules/sql/pages/querying-streams.adoc

```diff
@@ -97,91 +116,190 @@ If the client stops receiving results, but doesn't close the result, Hazelcast w
++-------------------+-------------------+--------------------+
+|2022-01-04T00:00   |2022-01-10T23:59   |                  13|
+|2022-01-05T00:00   |2022-01-11T23:59   |                  12|
+|2022-01-06T00:00   |2022-01-12T23:59   |                  13|
+```
+New results for each seven-day window are only returned when all events that belong to the window have been processed. To learn more about this, see <<late-events>>.
+
+[[late-events]]
+== Handling Late Events
+
+As stated in the previous section, Hazelcast can't emit the results of windowed aggregations or stream-to-stream joins until it has received all the events belonging to the defined timeframe. However, due to differences in latency, events that fall within the timeframe may not arrive for processing until after it ends. To place a limit on late events, Hazelcast uses the `IMPOSE_ORDER()` function. This function allows you to specify a _maximum event lag_. Any event that arrives later than the maximum event lag is dropped. 
 
 NOTE: Time is measured by the timestamps in the events, rather than the current time on a system clock.
 
-Hazelcast calls the function to add watermarks `IMPOSE_ORDER()`, because it curbs the potentially unbounded disorder of the events in the stream to a fixed value. The `IMPOSE_ORDER()` function is a stateful function whose state is scoped for the duration of the query. This function injects watermarks that lag a fixed amount behind the maximum value of the field observed since the query started.
+In the following example, the `IMPOSE_ORDER()` function injects a maximum event lag of 0.5 seconds for each aggregation window. An order event with a timestamp of `yyyy-mm-dd 23:59:59.5` is added to the window. If another event is processed with a timestamp that's 0.5 seconds or more old, such as ``yyyy-mm-dd 23:59:58.9`, that event is dropped because it is too old.
 
 ```sql
 SELECT *
 FROM TABLE(IMPOSE_ORDER(
-  TABLE trades, <1>
-  DESCRIPTOR(trade_time), <2>
+  TABLE orders, <1>
+  DESCRIPTOR(order_time), <2>
   INTERVAL '0.5' SECONDS) <3>
 );
 ```
 
 <1> The table that contains the events, including the timestamp.
 <2> A pointer to the column that contains the timestamp for the watermark.
-<3> The maximum event lag. Any events that are later than this lag are dropped. For example, an event with a timestamp of `yyyy-mm-dd 23:59:59.5` is added to the window. If another event is processed with a timestamp that's 0.5 seconds or more old, such as ``yyyy-mm-dd 23:59:58.9`, that event is dropped because it is too old.
-
-If an event is delayed by more than the defined maximum event lag, it is dropped.
+<3> The maximum event lag.
 
 NOTE: The above query doesn't currently run in Hazelcast. The `IMPOSE_ORDER()` function must be only used together with `TUMBLE` or `HOP` functions.
 
-For better readability, it's useful to create a view for the watermark like so:
+For better readability, it's useful to create a view:
```

Comment:
I'd make a suggestion to create a view for streaming source for aggregation/join queries as a good practice.

Reply:
Updated as requested.